torch/distributed/fsdp/_common_utils.py

```diff
@@ -266,11 +266,23 @@ def _get_param_to_fqns(
     dedup_shared_params: bool = True,
 ) -> Dict[nn.Parameter, List[str]]:
     """
-    Constructs a mapping from parameter to a list of its FQNs. Each normal
-    parameter maps to a singleton list containing its FQN, while each
+    Constructs a mapping from parameter to a list of its \"canonical\" FQNs. Here,
+    we use canonical to mean the fully-qualified name assigned to the parameter
+    based on its position in the nn.Module hierarchy. This is in contrast to FQNs
+    that may be generated by other parallelisms or wrappers.
```

Comment:
"generated" --> does this mean `named_parameters()`?

Reply:
yes, but i'm moreso referring to the fact that FlatParams get swizzled into the modules and have their own names